src/viam/components/resource_manager.py

```diff
@@ -64,6 +74,9 @@ def get_component(self, of_type: Type[ResourceType], name: ResourceName) -> Reso
         if component and isinstance(component, of_type):
             return component
 
+        if name.name in self._short_to_long_name and len(self._short_to_long_name[name.name]) == 1:
+            component = self.components.get(self._short_to_long_name[name.name][0], None)
+            return component
```

Comment:
`component` here could be `None`. You should either

1. Check that `component` is not `None` before returning OR
2. Return the result of recursively calling this function, but with the correct `ResourceName` (`return self.get_component(of_type, self._short_to_long_name[name.name][0])`)